Yosemite/Yosemite/Stores/OrderStore.swift

```diff
@@ -134,6 +142,19 @@ private extension OrderStore {
 }
 
 
+// MARK: - Private methods
+//
+private extension OrderStore {
+
+    /// Ensures Products are Synced for an Order.
+    ///
+//    func ensureProductsAreSynchronized(for order: Order, onCompletion: @escaping ([Product]?, Error?) -> Void) {
+//        let action = ProductAction.synchronizeProductsFor(order, onCompletion: onCompletion)
+//        dispatcher.dispatch(action)
+//    }
+}
```

Comment:
This code exists in case we decide that coupling the products remote requests to a successful GET Order by ID is the better solution.

Reply:
I don't think one store should know about the existence of other stores. Cross-cutting concerns should, to me, be resolved by another layer on top of the stores. 

This is something that sooner or later is going to come up again, as the app gets more complex it is likely there are going to be more dependencies between stores, so I think it's a discussion worth having sooner than later. 

In this particular case, I think we might be good if we used the OrderViewModel to be responsible of coordinating / managing that cross-cutting concern